os/binfmt/libelf/libelf_addrenv.c

```diff
@@ -105,7 +105,21 @@ static int allocateregions(FAR struct elf_loadinfo_s *loadinfo)
 		}
 	}
 
+#ifdef CONFIG_ARMV7M_MPU
+	/* ARMV7M requires addresses to be aligned to the size of the region.
+	 * In this case, we align the first region to the size of first region.
+	 * Since the regions are arranged in descending order of sizes and each
+	 * region size is a power of two, the remaining regions will also be
+	 * aligned to their size.
+	 */
 	*allocs[0] = (uintptr_t)kmm_memalign(sizes[0], totalsize);
+#elif CONFIG_ARMV8M_MPU
+	/* ARMV8M requires 32 bytes alignment of addresses */
+	*allocs[0] = (uintptr_t)kmm_memalign(MPU_ALIGNMENT_BYTES, totalsize);
```

Comment:
With v8m, I think we don't need to concatenate the regions for memory usage flexibility.
How about your opinion?

Reply:
True. I modified it.